x-pack/plugins/security_solution/public/common/components/user_privileges/endpoint/use_endpoint_privileges.ts

```diff
@@ -44,47 +40,21 @@ export const useEndpointPrivileges = (): Immutable<EndpointPrivileges> => {
   const fleetAuthz = fleetServicesFromUseKibana?.authz ?? fleetServicesFromPluginStart?.authz;
 
   const licenseService = useLicense();
-  const isPlatinumPlus = licenseService.isPlatinumPlus();
 
   const [userRolesCheckDone, setUserRolesCheckDone] = useState<boolean>(false);
   const [userRoles, setUserRoles] = useState<MaybeImmutable<string[]>>([]);
 
-  const isEndpointRbacEnabled = useIsExperimentalFeatureEnabled('endpointRbacEnabled');
-  const isEndpointRbacV1Enabled = useIsExperimentalFeatureEnabled('endpointRbacV1Enabled');
-
-  const [checkHostIsolationExceptionsDone, setCheckHostIsolationExceptionsDone] =
-    useState<boolean>(false);
-  const [hasHostIsolationExceptionsItems, setHasHostIsolationExceptionsItems] =
-    useState<boolean>(false);
-
   const privileges = useMemo(() => {
-    const loading = !userRolesCheckDone || !user || !checkHostIsolationExceptionsDone;
+    const loading = !userRolesCheckDone || !user;
```

Comment:
Below you removed the API call to check if HIE exist. I would prefer that logic remain here, as this is the only place we should be using to check if user can access it or not. I would rather not have specific implementations doing their own check on data.

Reply:
The check has been moved to links.ts to remove HIE link if needed, so all UI components on the HIE page are already gated by the check. It is explained in the comment below